catalyst/dl/runner/core.py

```diff
@@ -352,9 +352,23 @@ def trace(
                 self.device = utils.get_device()
             device = self.device
 
-        result = utils.trace_model(
+        # Dumping previous state of the model, we will need it to restore
+        _device, _is_training, _requires_grad = \
+            self.device, model.training, utils.get_requires_grad(model)
+
+        model.to(device)
+
+        # function to run prediction on batch
+        def predict_fn(model, inputs, **kwargs):
+            _model = self.model
+            self.model = model
+            result = self.predict_batch(inputs, **kwargs)
+            self.model = _model
+            return result
```

Comment:
could you please tell,
why do we need 
```
_model = self.model
self.model = model
```
and 
```
self.model = _model
```
here?

Reply:
`model` argument is a wrapper here, so I should change runner.model to wrapper, execute predict_batch and then reverse model back to original one, otherwise we will have wrapper of our model in runner :)